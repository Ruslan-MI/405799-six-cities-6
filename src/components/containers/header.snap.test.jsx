import React from "react";
import {
  render,
} from "@testing-library/react";
import Header from "./header";
import {
  AuthorizationStatus,
  StoreNameSpace,
} from "../../const";
import Test, {
  mockUserInitialState,
  mockAuthInfo,
} from "../../mocks-for-test";

describe(`Header should render correctly`, () => {
  it(`Header without authorization should render correctly`, () => {
    const dataForStore = {
      [StoreNameSpace.USER]: mockUserInitialState,
    };

    const {
      container,
    } = render(<Test dataForStore={dataForStore}>
      <Header />
    </Test>);

    expect(container).toMatchSnapshot();
  });

  it(`Header with authorization should render correctly`, () => {
    const dataForStore = {
      [StoreNameSpace.USER]: {
        ...mockUserInitialState,
        authorizationStatus: AuthorizationStatus.AUTH,
        userEmail: mockAuthInfo.email,
      },
    };

    const {
      container,
    } = render(<Test dataForStore={dataForStore}>
      <Header />
    </Test>);

    expect(container).toMatchSnapshot();
  });
});
