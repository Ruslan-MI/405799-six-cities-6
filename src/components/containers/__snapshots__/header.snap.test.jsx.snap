// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly Header with authorization should render correctly 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link"
            href="/"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
            style="flex-direction: column; align-items: flex-end;"
          >
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/favorites"
              >
                <div
                  class="header__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="User avatar"
                    class="user__avatar"
                    height="74"
                    src=""
                    width="74"
                  />
                </div>
                <span
                  class="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link"
                href="/"
              >
                <span>
                  Logout
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header should render correctly Header without authorization should render correctly 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link"
            href="/"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
            style="flex-direction: column; align-items: flex-end;"
          >
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/login"
              >
                <div
                  class="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  class="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</div>
`;
